src/app/dataset/depublication/depublication.component.ts

```diff
@@ -92,13 +92,43 @@ export class DepublicationComponent implements OnDestroy {
   */
   buildForms(): void {
     this.formRawText = this.fb.group({
-      recordIds: ['', [Validators.required, this.validateWhitespace]]
+      recordIds: [
+        '',
+        [Validators.required, this.validateWhitespace, this.validateRecordIds.bind(this)]
+      ]
     });
     this.formFile = this.fb.group({
       depublicationFile: ['', [Validators.required, this.validateFileExtension]]
     });
   }
 
+  /** validateRecordIds
+  /*  returns an error if the form control value includes invalid record ids
+  /*  @param {FormControl} control - the input control to validate
+  */
+  validateRecordIds(control: FormControl): { [key: string]: boolean } | null {
+    const val = control.value || '';
+    let invalid = false;
+    let misref = false;
+
+    const reg = /(http(s)?:\/\/)?([a-zA-Z]*\/)?([0-9]*\/)?[a-zA-Z\d_]*$/;
+    const reg2 = new RegExp('\\b' + this._datasetId + '\\b');
```

Comment:
And here we should ensure that the database id occurs at the right segment. How about capturing it as a group in the first regular expression and verifying it? Building on my previous comment, it would look like this:
`/^((http(s)?:\/\/)?([^\s\/]+\/)*(?<dataset_capture>[a-zA-Z\d]+)\/)?\w+$/`
And then we check the value of `dataset_capture` and see if it equals `this._datasetId` (if it even occurs).